### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by uwadonat**
### Math Magicians: refactor with hooks

**In this milestone I implemented the following:**

- Refactor the Calculator component (components/Calculator.js) from a class based component to a functional component with hooks.
- Testing the app in the browser

**Hope to meet all requirements of this milestone**

_Thank you for your time to review my project_

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,46 +1,32 @@
...
-    this.state = {};
-    this.handleClick = this.handleClick.bind(this);
-  }
+function Calculator() {
+  const [calcObject, setCalcObject] = useState({});
 
-  handleClick(e) {
-    this.setState((state) => calculate(state, e.target.textContent));
+  function handleClick(e) {
+    setCalcObject({ ...calcObject, ...calculate(calcObject, e.target.textContent) });
```
Comment (henatan99) [inline src/components/Calculator.js]: - [x] I think you need to implement error handling method or condition. When an operator button is clicked twice or repeatedly it throws an error on the browser. Kindly fix this by implementing error handling. :pray:

Screenshot from 2021-08-20 15-41-15

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,46 +1,32 @@
...
 import './Calculator.css';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-    this.handleClick = this.handleClick.bind(this);
-  }
+function Calculator() {
```
Comment (henatan99) [inline src/components/Calculator.js:5]: - [ ] OPTIONAL: I recommend using arrow functions as it is today's common practice to use them whenever possible. You may make changes if youfind this comment relevant.

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @uwadonat , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (uwadonat) [reply src/components/Calculator.js]: error handling was implemented.

**Review by ZawadiSumba66 — APPROVED** [review]
Hi @uwadonat 👋  , 
## STATUS: APPROVED 👍 
Great job implementing the changes requested by the previous reviewer. 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**